src/deye_daemon.py

```diff
@@ -109,6 +117,36 @@ def __remove_duplicated_reg_ranges(self, reg_ranges: list[SensorRegisterRange])
                 result.append(reg_range)
         return result
 
+    def __new_event(self, events: list[DeyeEvent]) -> bool:
+        """Check if the received event data has changed compared
+
+        Parameters
+        ----------
+        events : list[DeyeEvent]
+            Received event list
+
+        Returns
+        -------
+        bool
+            True if received events is different from last or new events received (offline)
+        """
+
+        if not events[0]:  # offline
+            self.__log.debug("No events received (offline)")
+            return True
+        # Compare list of received events with last observation, ignoring status changes
+        if (
+            compare_event_list(events, self.__last_observations, check_status=False)
+            and self.__event_updated + self.__event_expiry > time.time()
+        ):
+            self.__log.debug("Event data is unchanged and hasn't expired")
+            return False
+        self.__log.debug("Time since previous update: %s", time.time() - self.__event_updated)
+        self.__log.debug("New event data: %s", [str(e) for e in events])
+        self.__event_updated = time.time()
+        self.__last_observations = events[1:]  # ignore status field
```

Comment:
Why `events[0]` is ignored?

Reply:
agree that this is not needed, changed to full events list